tqecjs/src/workspace/QubitClass.js

```diff
@@ -0,0 +1,130 @@
+import { Graphics, Text } from 'pixi.js';
+
+/**
+ * Qubit class
+ * @extends Graphics
+ * @constructor
+ * @param {number} x - The x position of the qubit
+ * @param {number} y - The y position of the qubit
+ */
+export default class Qubit extends Graphics {
+	constructor(x, y, radius = 5, color = 0x000000, gridSize = 50) {
+		super();
+		// UI properties
+		this.eventMode = 'static';
+		this.buttonMode = true;
+		this.cursor = 'pointer';
+		this.globalX = x;
+		this.globalY = y;
+		this._createCircle(x, y, radius, color);
+		this.maxNeighborDist = 2 * this.gridSize;
+		this.neighbors = [];
+		this.gridSize = gridSize;
+		// Adjacent (degree 1) qubits
+		// QC properties
+		this.quantumState = '0'; // NOTE: In javascript, becareful updating values such as "this", "state", etc.
+		this.isQubit = true;
+	}
+	_onPointerOver = () => {
+		this.alpha = 0.5;
+	};
+	_onPointerOut = () => {
+		this.alpha = 1;
+	};
+	/**
+	 * Creates a circle
+	 * @param {*} x
+	 * @param {*} y
+	 * @param {*} radius
+	 * @param {*} color
+	 */
+	_createCircle(x, y, radius, color) {
+		// Create a circle
+		this.beginFill(color);
+		this.drawCircle(x, y, radius);
+		// this.hitArea = new Circle(x, y, radius);
+		this.endFill();
+
+		// Add hover event
+		this.on('pointerover', this._onPointerOver);
+		this.on('pointerout', this._onPointerOut);
+	}
+
+	checkHitArea(eventX, eventY, threshold = 5) {
+		// Calculate the distance between event coordinates and qubit's global position
+		const distance = Math.sqrt(
+			Math.pow(eventX - this.globalX, 2) + Math.pow(eventY - this.globalY, 2)
```

Comment:
Might be good to have a general `distance()` function

Reply:
sg

Reply:
the code will probably require another round of refactoring and ill leave it for it to be done then